apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/DbMultiDomainAcsStore.scala

```diff
@@ -231,9 +231,27 @@ final class DbMultiDomainAcsStore[TXE](
 
   def hasArchived(ids: Seq[ContractId[?]])(implicit
       traceContext: TraceContext
-  ): Future[Boolean] =
-    // TODO(#838): implement this as a single DB query
-    Future.sequence(ids.map(lookupContractStateById)).map(_.exists(_.isEmpty))
+  ): Future[Boolean] = waitUntilAcsIngested {
+    if (ids.isEmpty) Future.successful(true)
+    else {
+      storage.query(
+        sql"""
+        select not exists (
+           select acs.contract_id
+           from #$acsTableName acs
+           where acs.store_id = $acsStoreId
+           and acs.migration_id = $domainMigrationId
+           and acs.contract_id in (#${ids
+            .map(id => lengthLimited(s"'${id.contractId}'"))
```

Comment:
This is not great, I'll check if there is a better way with Slick to do this instead of raw, manual interpolation.

Reply:
Can you use `inClause`?

https://github.com/hyperledger-labs/splice/blob/03bbae02019515d3b307f46a7213752001f0f59b/apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/AcsQueries.scala#L259-L260

https://github.com/hyperledger-labs/splice/blob/a82fa0d82c19d12d195e40575525022160fbcfc6/apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/AcsSnapshotStore.scala#L217-L219

Reply:
Oh nice, I probably can! It is really safer though? I guess this binds properly through SetParameter?

Reply:
Exactly.